compiler/AST/primitive.cpp

```diff
@@ -715,3 +716,12 @@ VarSymbol* newMemDesc(const char* str) {
   memDescsVec.add(s);
   return memDescVar;
 }
+
+VarSymbol* newMemDesc(Symbol* sym) {
```

Comment:
When newMemDesc is invoked from callChplHereAlloc(), it is passed a Symbol, not a TypeSymbol. I doubt it is worth converting Symbol to TypeSymbol in that case.